feature/foryou/src/main/java/com/google/samples/apps/nowinandroid/feature/foryou/ForYouScreen.kt

```diff
@@ -126,23 +123,8 @@ internal fun ForYouScreen(
     val isOnboardingLoading = onboardingUiState is OnboardingUiState.Loading
     val isFeedLoading = feedState is NewsFeedUiState.Loading
 
-    // Workaround to call Activity.reportFullyDrawn from Jetpack Compose.
-    // This code should be called when the UI is ready for use
-    // and relates to Time To Full Display.
-    // TODO replace with ReportDrawnWhen { } once androidx.activity-compose 1.7.0 is used (currently alpha)
-    if (!isSyncing && !isOnboardingLoading && !isFeedLoading) {
-        val localView = LocalView.current
-        // We use Unit to call reportFullyDrawn only on the first recomposition,
-        // however it will be called again if this composable goes out of scope.
-        // Activity.reportFullyDrawn() has its own check for this
-        // and is safe to call multiple times though.
-        LaunchedEffect(Unit) {
-            // We're leveraging the fact, that the current view is directly set as content of Activity.
-            val activity = localView.context as? Activity ?: return@LaunchedEffect
-            // To be sure not to call in the middle of a frame draw.
-            localView.doOnPreDraw { activity.reportFullyDrawn() }
-        }
-    }
+    // This code should be called when the UI is ready for use and relates to Time To Full Display.
+    ReportDrawnWhen { !isSyncing && !isOnboardingLoading && !isFeedLoading }
```

Comment:
Is there any downside for this never being called, or being called much later in the case where we aren't on the for you screen?

For instance, if we navigate to another tab, and then rotate the device, we won't call `ReportDrawnWhen` until we navigate back to the for you screen later.

Reply:
If this is never called, system assumes fully drawn ~5seconds after Activity.onCreate.

There isn't any downside at this point.

If there were multiple entrypoints into the app (e.g deeplinks), we should implement that there as well.

Reply:
Great info to know, thanks!